mm2src/mm2_p2p/src/behaviours/atomicdex.rs

```diff
@@ -162,6 +182,22 @@ pub enum AdexBehaviourCmd {
     },
 }
 
+/// Determines if a dial attempt to the remote should be made.
+///
+/// Returns `false` if a dial attempt to the given address has already been made,
+/// in which case the caller must skip the dial attempt.
+fn pre_dial_check(recently_dialed_peers: &mut MutexGuard<TimedMap<StdClock, Multiaddr, ()>>, addr: &Multiaddr) -> bool {
```

Comment:
Neither the doc comment nor the func name signify how this function changes the timedmap (only visible from the mutability of the passed map). Let's change this to something like `check_and_mark_dialed` or something.

Also, no need for the `MutexGuard` annotation here.

Reply:
> Neither the doc comment nor the func name signify how this function changes the timedmap (only visible from the mutability of the passed map)

"only visible from the mutability of the passed map" that's visible everywhere, both in the function signature and on the caller side!

I struggle to see how `check_and_mark_dialed` describes the function better than `pre_dial_check` but whatever, I am okay with it.

> Also, no need for the MutexGuard annotation here.

It gives/documents the background context how this map is utilized from caller.